### ISSUE TITLE
New rules: selector-no-compound-with-id/type

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#428]
* [ ] `selector-no-compound-with-id` - Disallow compound selectors that contain _id selectors_.
* [ ] `selector-no-compound-with-type` - Disallow compound selectors that contain _type selectors_.

**Comment by davidtheclark** [comment]
Here's the big question with this one: How rigid are we about which qualifications?

Should we make it `selector-no-chained-id`? That would mean that you're never allowed to have a id simple selector chained to any other selector within a compound selector. That's the type of qualification that I think never really has a good justification (except to override some existing horrible, horrible mess, maybe).

We could also make another rule like `selector-no-chained-type`, as I think some people would want to restrict that in their codebase. I'm not sure, though, about real use-cases for restricting the chaining of classes or pseudos or attributes.

What do you think?

**Comment by jeddy3** [comment]
Eeek. You're right, that's a big question!

I think also think we're right to make things specific.

Both those use-cases do make sense and I agree that there isn't really a use-case for restricting _class_, _pseudos_ and _attribute_ selectors.

I'm wondering if the rule names need tweaking as I didn't get the `-chained-` bit at first. Shall we tied them to _compound selectors_ e.g.

```js
"selector-no-compound-with-id"
"selector-no-compound-with-type"
```

Or something along those lines. At least that way we can point to the [spec](https://drafts.csswg.org/selectors-4/#syntax) in the docs for those who don't know what a _compound selector_ is.

**Comment by davidtheclark** [comment]
`chained` is actually used to define "compound" selectors in Selectors Level 4 and to define "sequence" in Selectors Level 3. So either way we're going with spec language. I'm indifferent, though: happy to go with what you like best.

**Comment by jeddy3** [comment]
> chained is actually used to define "compound" selectors in Selectors Level 4

I didn't know that. It's good to know :)

> happy to go with what you like best.

Lets rock `selector-no-compound-with-id/type` as it's consistent with #431.

**Comment by davidtheclark** [comment]
Still not sure about this one, actually. There are edge cases. Like do we allow chained attributes with types? That's pretty common. And what about state class on an id selector, e.g. `#foo.is-open`, `#foo.is-closed`. I'm not advocating these practices; but I just wonder if this rule would be too cut-and-dry to be real-world useful.

**Comment by jeddy3** [comment]
> but I just wonder if this rule would be too cut-and-dry to be real-world useful.

Shall we kill this issue for now and wait to see what real-world requests come up?

**Comment by davidtheclark** [comment]
I think so.
